### ISSUE TITLE
Meaningful link text in spec

### ISSUE THREAD

**Issue by michael-n-cooper** [w3c/wcag21#158]
Filed by private email

There are two links on the page that do not have meaningful link text.  They are:

        <a id="toc-jump" href="#toc"><span aria-hidden="true">↑</span> <span>Jump to Table of Contents</span></a>

        and

        <a id="toc-toggle" href="#toc"><span aria-hidden="true">→</span> <span>Pop Out Sidebar</span></a>

        When I look at the code...my brain thinks..."I see link text...it should be fine."  But when I tried it in VoiceOver, NVDA and JAWS...alas....the link text is not good in any of these 3 screen readers.

**Comment by awkawk** [comment]
What do the screen readers voice for these links?

**Comment by steverep** [comment]
NVDA in Firefox will read differently based on whether those links are tabbed to or arrowed to and things also change depending on whether the mouse is hovering them.  IMHO, this issue goes way beyond meaningful link text though.  The whole TOC mechanism is not very screen reader friendly.

For example, the collapse and pop-out sidebar link should really be a button and marked with `aria-expanded` like an accordion.  Then it gets confusing because you can still jump to the TOC which has moved into the main flow under the abstract... In short, I only navigate these Respec documents via heading because the TOC is confusing (and there's no main landmark so that doesn't help either).

**Comment by michael-n-cooper** [comment]
This issue is complicated for us because the problems come from sources we do not control. 

The statement "The whole TOC mechanism is not very screen reader friendly" may relate primarily to the [W3C TR styles](https://github.com/w3c/tr-design), which all TR documents are required to use. We can attempt to override some of the styles, but there are rules about that and publication will be refused if we break them. We should file issues on the TR styles repository and request to have the styles update, that will likely be the primary fix we can obtain but it won't come quickly.

Some of the specific problems noted in the original issue may come from how [Respec](https://github.com/w3c/respec/) instantiates the TR design. For that latter we can do some things by manual post-processing, but I'm anxious to minimize the amount of work involved in that. I would need a concrete list of things we might address that way. I would need a specific edit-list of things to add to my "fix bad stuff Respec does" list.

**Comment by steverep** [comment]
@michael-n-cooper, I totally get that this is not a WCAG issue but one that affects all TR documents, so we definitely should file these issues in the appropriate repository (as with the above Respec issues I referenced regarding permalinks inside heading tags).

However, we should not just accept that such issues won't get fixed quickly.  The irony of producing a WCAG publication that wouldn't pass an accessibility audit is too great.

**Comment by jake-abma** [comment]
The toc-nav is indeed unreliable and doesn’t deserve the beauty award.
Here the results where especially the link lists and arrow keys may cause weird feedback.

We may have an exemption if it’s not under our own control but also an obligation to put some friendly pressure on who does.

**Firefox / NVDA mentions in…**

Link list:
- ↑ 1 of 809 level 0
- → 2 of 809 level 0

Tab key:
- Jump to Table of Contents link
- Pop Out Sidebar link

Arrow key:
- Jump to Table of Contents link
- Pop Out Sidebar link

**InternetExplorer / Jaws mentions in…**

Link list:
- #toc
- #toc

Tab key:
- Jump to Table of Contents same page link
- Pop Out Sidebar same page link

Arrow key:
- same page link number toc
- same page link number toc

**Safari / Chrome / VoiceOver mentions in…**

Link list:
- Link WCAG21
- Link WCAG21

Tab key:
- Link Jump to Table of Contents
- Link Pop Out Sidebar

Arrow key:
- Link Jump to Table of Contents
- Link Pop Out Sidebar

**Comment by awkawk** [comment]
Not a WCAG 2.1 issue but a respectable issue. @michael-n-cooper is this one filed with respect?

**Comment by michael-n-cooper** [comment]
This is TR design not Respec. I filed [Issue 136](https://github.com/w3c/tr-design/issues/136) there, closing here.
